railgun/internal/diagnostics/server.go

```diff
@@ -0,0 +1,54 @@
+package diagnostics
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/kong/kubernetes-ingress-controller/railgun/internal/mgrutils"
+	"github.com/kong/kubernetes-ingress-controller/railgun/pkg/config"
+	"github.com/sirupsen/logrus"
+)
+
+type diagnosticsServer struct {
+	logger logrus.FieldLogger
+	mux    *http.ServeMux
+}
+
+func NewDiagnosticsServer(enableProfiling bool, log logrus.FieldLogger) *diagnosticsServer {
+	mux := http.NewServeMux()
+	if enableProfiling {
+		mgrutils.Install(mux)
+	}
+	return &diagnosticsServer{
+		logger: log,
+		mux:    mux,
+	}
+}
+
+func (s *diagnosticsServer) Start(ctx context.Context) error {
```

Comment:
This function blocks for the lifetime of the profiling server, while `Start` implies that this function just "starts" the server (in the background).

Please rename to `Listen`.